src/plugins/initiatives/createGraph.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+
+import {
+  Graph,
+  EdgeAddress,
+  NodeAddress,
+  type Edge,
+  type Node,
+  type EdgeAddressT,
+  type NodeAddressT,
+} from "../../core/graph";
+import type {Initiative, URL, InitiativeRepository} from "./initiative";
+import {
+  initiativeNodeType,
+  dependsOnEdgeType,
+  referencesEdgeType,
+  contributesToEdgeType,
+  championsEdgeType,
+} from "./declaration";
+
+// TODO: this should probably be in the core, when universal reference detection is added.
+/**
+ * A service which provides reference detection features.
+ */
+export interface ReferenceDetector {
+  /**
+   * Tries to infer the node address from an absolute URL.
+   * Returning undefined when the detector isn't aware of how to resolve this URL.
+   * Note: the detector shouldn't attempt to detect whether the node actually exists.
+   */
+  addressFromUrl(url: URL): ?NodeAddressT;
+}
```

Comment:
This TODO now has it's own PR #1508
So I plan to revise this PR to depend on it.

Reply:
Did exactly that, depending on the core reference detector interface now.